### PR TITLE
allow user flows when the acting user is legitimate, but not its owner

### PR DESCRIPTION
**Pull request by blizzz**
for instance, when a sharee changes a file, the owner can act upon

To Dos:
* [x] expose the owner of the operation to act (e.g. when writing to conversations)

### CODE REVIEW

File: lib/composer/composer/autoload_classmap.php [hunk 1]
```diff
@@ -445,6 +445,10 @@
     'OCP\\User\\Backend\\ISetPasswordBackend' => $baseDir . '/lib/public/User/Backend/ISetPasswordBackend.php',
     'OCP\\User\\Events\\PostLoginEvent' => $baseDir . '/lib/public/User/Events/PostLoginEvent.php',
     'OCP\\Util' => $baseDir . '/lib/public/Util.php',
+    'OCP\\WorkflowEngine\\EntityContext\\IDisplayName' => $baseDir . '/lib/public/WorkflowEngine/EntityContext/IDisplayName.php',
```
Comment (juliushaertl) [inline lib/composer/composer/autoload_classmap.php]: Files are missing :wink:

Comment (blizzz) [reply lib/composer/composer/autoload_classmap.php]: m(

**Comment by blizzz** [comment]
Tests also need https://github.com/nextcloud/server/pull/18143 it seems

**Comment by rullzer** [comment]
> Tests also need #18143 it seems

in

**Comment by juliushaertl** [comment]
@blizzz I pushed another commit to set the file info in the rule matcher. That was missing when I tested it with the talk operation. Please give it a quick sanity check. Otherwise looks good to me and works as advertised.

**Review by juliusknorr — APPROVED** [review]
🚀

**Comment by blizzz** [comment]
remaining failing  tests are irrelevant to this PR

**Review by rullzer — APPROVED** [review]
Looks good!

**Pull request merged**